GenericJSON/Querying.swift

```diff
@@ -0,0 +1,70 @@
+import Foundation
+
+public extension JSON {
+
+    /// Return the string value if this is a `.string`, otherwise `nil`
+    public var stringValue: String? {
+        if case .string(let value) = self {
+            return value
+        }
+        return nil
+    }
+
+    /// Return the float value if this is a `.number`, otherwise `nil`
+    public var floatValue: Float? {
+        if case .number(let value) = self {
+            return value
+        }
+        return nil
+    }
+
+    /// Return the bool value if this is a `.bool`, otherwise `nil`
+    public var boolValue: Bool? {
+        if case .bool(let value) = self {
+            return value
+        }
+        return nil
+    }
+
+    /// Return the object value if this is an `.object`, otherwise `nil`
+    public var objectValue: [String: JSON]? {
+        if case .object(let value) = self {
+            return value
+        }
+        return nil
+    }
+
+    /// Return the array value if this is an `.array`, otherwise `nil`
+    public var arrayValue: [JSON]? {
+        if case .array(let value) = self {
+            return value
+        }
+        return nil
+    }
+
+    /// Return `true` iff this is `.null`
+    public var isNull: Bool {
+        if case .null = self {
+            return true
+        }
+        return false
+    }
+
+    /// If this is an `.array`, return item at index
+    ///
+    /// If this is not an `.array` or the index is out of bounds, returns `nil`.
+    public subscript(index: Int) -> JSON? {
+        if case .array(let arr) = self, index < arr.count {
```

Comment:
We check the upper bound of the array here, does it make sense to check the lower bound (i.e. 0) too?

Reply:
Good catch! Will do.